frontend/src/employee-frontend/components/messages/MessageEditor.tsx

```diff
@@ -187,20 +195,33 @@ export default React.memo(function MessageEditor({
   )
   useEffect(
     function updateReceiversOnSenderChange() {
+      const previousReceivers = message.recipientIds
```

Comment:
IMHO it is wrong to read from `message.recipientIds` here since its value is deduced from the `receiverTree`. This creates a nasty two-way data flow.
I believe you could/should use the previous value provided by `setReceiverTree` function and get rid of the `!isEqual()` check altogether.

Reply:
Thanks. Refactored as suggested.